src/python/pants/backend/native/subsystems/native_build_step_settings.py

```diff
@@ -30,9 +32,19 @@ def register_options(cls, register):
              help='The default for the "compiler_option_sets" argument '
                   'for targets of this language.')
 
+    register('--toolchain-variant', type=str, fingerprint=True, advanced=True,
+             choices=ToolchainVariant.allowed_values,
+             default=ToolchainVariant.default_value,
+             help="Whether to use gcc (gnu) or clang (llvm) to compile C and C++. Currently all "
+                  "linking is done with binutils ld on Linux, and the XCode CLI Tools on MacOS.")
+
   def get_compiler_option_sets_for_target(self, target):
     return self.get_target_mirrored_option('compiler_option_sets', target)
 
+  @property
+  def toolchain_variant(self):
+    return ToolchainVariant.create(self.get_options().toolchain_variant)
```

Comment:
Could you please convert this into a method similar to `get_compiler_option_sets_for_target(self, target)` here? If we turn this into a method, we get [the logic you're doing in `native_task.py`](https://github.com/pantsbuild/pants/pull/7179/files#diff-6472aee7b82f9f7014f21cf71dd68f4dR95) for free:
```python
    def get_toolchain_variant_for_target(self, target):
      return ToolchainVariant.create(self.get_target_mirrored_option('toolchain_variant', target))
```
(I've left a further explanatory comment in `native_task.py`)